syntax = "proto3";

message None {
}

message CoprocReq {
  oneof payload {
    SetLeds setLeds = 4;
    GetButtons getButtons = 5;
    SetStupidServo setStupidServo = 6;
  }

  message SetLeds {
    LedsEnum ledsOn = 1;
  }

  enum LedsEnum {
    NONE = 0;
    L1 = 1;
    L2 = 2;
    L3 = 4;
    L4 = 8;
  }

  message GetButtons {
  }

  message SetStupidServo {
    uint32 servoIndex = 1;
    oneof servoCmd {
      None disable = 4;
      float setPosition = 5;
    }
  }
}

message CoprocStat {
  oneof payload {
    None ledsStat = 4;
    ButtonsStat buttonsStat = 5;
    None stupidServoStat = 6;
  }

  message ButtonsStat {
    ButtonsEnum buttonsPressed = 1;
  }

  enum ButtonsEnum {
    NONE = 0;
    BOFF = 1;
    B1 = 2;
    B2 = 4;
    B3 = 8;
    B4 = 16;
    BON = 32;
  }
}
